security/custom-gas-token.md

```diff
@@ -0,0 +1,83 @@
+
+Below are references for this project:
+
+- [Custom Gas Token - Product Hub](https://www.notion.so/Custom-Gas-Token-Product-Hub-4eeb15de1c4545b5bef68fc24b573a71?pvs=21)
+- [[EXTERNAL] How to run a Custom Gas Token chain](https://www.notion.so/EXTERNAL-How-to-run-a-Custom-Gas-Token-chain-d88c68306f934790be051f87213a0b1d?pvs=21)
+- https://github.com/ethereum-optimism/specs/blob/08d3c7189309e9e739d11312ed2ac7c1a43cb582/specs/protocol/granite/custom-gas-token.md
+- [https://github.com/ethereum-optimism/specs/discussions/140](https://github.com/ethereum-optimism/specs/discussions/140)
+- https://github.com/ethereum-optimism/optimism/pull/10143
+
+## **Failure Modes and Recovery Paths**
+
+### Bridge Undercollateralization
+
+- **Description**
+    - It is possible that the bridge becomes undercollateralized. This means that the invariant is broken that the bridge owns greater than or equal to custom gas token on L1 compared to the amount of native asset minted on L2. If the bridge becomes undercollateralized, then users that deposited into L2 will not be able to withdraw. If the chain can interoperate with other chains, then this can impact the collateralization of the entire superchain.
+- **Risk Assessment:**
+    - High severity, low likelihood
+    - A hack that steals money from the bridge can end the chain. This is not battle tested code but follows best practices like checks, effects, interactions, has thorough unit and end to end tests as well as follows a similar pattern to Fraxtal’s implementation of custom gas token. Fraxtal has not yet been hacked.
+    - A hack that mints money on L2 without it being collateralized by the rollup should not be possible given a standard ERC20 token is used. This is certainly possible to do with a malicious ERC20 token and this is not possible to detect on chain. This means that social consensus must be used to determine the safety of a rollup using a custom gas token.
+    - Another form of this attack involves using `ether` to mint native asset on L2 when the custom gas token feature is enabled. This would undercollaterize the bridge in terms on the custom gas token and turn the native asset into a proportionate representation of `ether` and the custom gas token. This would break withdrawals.
+- **Mitigations:**
+    - This can be mitigated by using offchain checks on the standardness of the ERC20 token before doing any sort of brand association with the custom gas token chain. These checks cannot be automated, manual inspection must be done to ensure that the token is [specs compliant](https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/granite/custom-gas-token.md#properties-of-a-gas-paying-token).
+- **Detection:**
+    - An offchain monitoring service could exist that is able to utilize the APIs on the smart contracts to determine that the bridge is not drained on L1 in a malicious way
```

Comment:
Have we implemented such a service already?

Reply:
`wd-mon` does cover some but not all possible bridge drains